internal/translation/searchindex/searchindex.go

```diff
@@ -323,12 +323,14 @@ func (s *SearchIndex) Normalize() (*SearchIndex, error) {
 }
 
 // toAtlas converts internal SearchIndex representation to the Atlas structure used for API calls
-func (s *SearchIndex) toAtlas() (*admin.ClusterSearchIndex, error) {
-	convertJSONToListOfMaps := func(in *apiextensionsv1.JSON) (*[]map[string]interface{}, error) {
+//
+//nolint:dupl
```

Comment:
This here might indicate a code smell, such as we are duplicating code that could be avoided. Probably sliced into smaller reusable functions?

Reply:
In general the idiom `funcName := func(...) {...}` to make a closure does not usually pay off, you can always add the argument to de function and make it a regular internal function of the module. That also makes the caller method shorter and easier to read.

In this example I would just have instead elsewhere:
```go
func convertJSONToListOfMaps(in *apiextensionsv1.JSON /*add any extra args needed here*/) (*[]map[string]interface{}, error) {
...
}
```

Why are the named closures a better fit here?
Hopefully they are not repeated across functions, cause in that case all the more reason to define them once externally.